src/test/java/test/listeners/GitHub911Sample.java

```diff
@@ -0,0 +1,20 @@
+package test.listeners;
+
+import org.testng.Assert;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+public class GitHub911Sample {
+
+  @BeforeSuite(alwaysRun = true)
+  public void setUp() {
+    Assert.fail();
+  }
+  // TODO check before invocation
```

Comment:
@cbeust Currently, `beforeInvocation` and `afterInvocation` are called even if a configuration method will fail and tests will be skipped.
The javadoc is not really verbose.
In one hand, the behavior is not very logic but in other hand, it is the easier way to be notified by skipped method.